ed_sensor_integration/src/kinect/image_buffer.cpp

```diff
@@ -158,9 +158,20 @@ bool ImageBuffer::getMostRecentImageTF()
                                 << ros::Time(rgbd_image->getTimestamp()));
                 ROS_WARN_STREAM("[IMAGE_BUFFER] Image too old to look-up tf: image timestamp = " << std::fixed
                                 << ros::Time(rgbd_image->getTimestamp()));
+                return false;
+            }
+            else
+            {
+                if(!tf_listener_->waitForTransform(root_frame_, rgbd_image->getFrameId(), ros::Time(rgbd_image->getTimestamp()), ros::Duration(0.5)))
```

Comment:
With this you cannot guarantee the execution rate of 20Hz(or whatever you initialised it to) of your worker thread. It also means you get the same problem if tf lags by 0.5seconds. 
And this is a magic number.